File: javascript/smartlist/smartlist-api-mysql-project/src/modules/candidate/dto/response/candidate-assessment.resp.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsInt, IsNotEmpty } from 'class-validator';
import { CandidateAssessment } from 'src/entities/candidate-assessment.entity';
import { AssessmentDecision } from 'src/enums/assessment.decision';

export class CandidateAssessmentRespDto {
  @ApiProperty({
    type: Number,
  })
  @IsNotEmpty()
  @IsInt()
  id: number;

  @ApiProperty({
    type: Number,
  })
  @IsNotEmpty()
  @IsInt()
  candidate_id: number;

  @ApiProperty({
    type: Number,
  })
  @IsNotEmpty()
  @IsInt()
  assessment_id: number;

  @ApiProperty({
    type: Number,
  })
  @IsNotEmpty()
  @IsInt()
  active_assessment_block_id: number;

  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  status: string;

  @ApiProperty({
    type: Date,
  })
  @IsNotEmpty()
  start_date: Date;

  @ApiProperty({
    type: Date,
  })
  end_date: Date;

  @ApiProperty({
    type: String,
  })
  assessment_decision: AssessmentDecision;

  constructor(candidateAssessment: CandidateAssessment) {
    this.id = candidateAssessment.id;
    this.candidate_id = candidateAssessment.candidate.id;
    this.assessment_id = candidateAssessment.assessment.id;
    this.active_assessment_block_id = candidateAssessment.assessmentBlock.id;
    this.status = candidateAssessment.status;
    this.start_date = candidateAssessment.startDate;
    this.end_date = candidateAssessment.endDate;
    this.assessment_decision = candidateAssessment.assessmentDecision;
  }
}

File: java/smartlist/src/main/java/com/smartlist/module/candidate/dto/response/CandidateAssessmentRespDTO.java
package com.smartlist.module.candidate.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.smartlist.enums.AssessmentDecision;
import com.smartlist.model.Assessment;
import com.smartlist.model.CandidateAssessment;
import com.smartlist.module.assessment.dto.response.AssessmentRespDTO;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Getter;
import lombok.Setter;
import org.modelmapper.ModelMapper;
import org.modelmapper.TypeMap;

import java.time.LocalDateTime;
import java.util.Date;

@Getter
@Setter
public class CandidateAssessmentRespDTO {

    @Schema(description = "Candidate Assessment Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    Integer id;

    @Schema(description = "Candidate Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    @JsonProperty(value = "candidate_id")
    Integer candidateId;

    @Schema(description = "Assessment Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    @JsonProperty(value = "assessment_id")
    Integer assessmentId;

    @Schema(description = "Active Assessment Block Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    @JsonProperty(value = "active_assessment_block_id")
    Integer assessmentBlockId;

    @Schema(description = "Status", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotBlank
    String status;

    @Schema(description = "Start Date", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotEmpty
    @JsonProperty(value = "start_date")
    LocalDateTime startDate;

    @Schema(description = "End Date", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotEmpty
    @JsonProperty(value = "end_date")
    LocalDateTime endDate;

    AssessmentDecision assessmentDecision;

}
